src/main/java/com/global/api/gateways/Gateway.java

```diff
@@ -171,7 +182,7 @@ private String buildQueryString(HashMap<String, String> queryStringParams) throw
     }
     
     private String parseErrorMessage(String rawResponse) throws ApiException {
-    	ElementTree tree = ElementTree.parse(rawResponse);
+        ElementTree tree = ElementTree.parse(rawResponse);
         Element root = tree.get("soap:Fault");
```

Comment:
You should probably also check that root is not null before attempting to read any of its children since this could be an NRE if the XML response is not soap based. Which is entirely possible.

Reply:
I reservation was more about the other implementations that will be required in the other subclasses. Unless you want me to leave the project in a uncompiling state. I could just make the other implementations just return the input rawResponse, but that probably isn't very helpful.

Reply:
Which is fine for now. I can take it from there.